mesh/mesh.cpp

```diff
@@ -3380,6 +3541,271 @@ void Mesh::Make3D(int nx, int ny, int nz, Element::Type type,
    // Finalize(...) can be called after this method, if needed
 }
 
+
+void Mesh::Make2D4TrisFromQuad(int nx, int ny, double sx, double sy)
+{
+   SetEmpty();
+
+   Dim = 2;
+   spaceDim = 2;
+
+   NumOfVertices = (nx+1) * (ny+1);
+   NumOfElements = nx * ny * 4;
+   NumOfBdrElements =  (2 * nx + 2 * ny);
+   vertices.SetSize(NumOfVertices);
+   elements.SetSize(NumOfElements);
+   boundary.SetSize(NumOfBdrElements);
+   NumOfElements = 0;
+
+   double cx, cy;
+   int ind[4];
+
+   // Sets vertices and the corresponding coordinates
+   int k = 0;
+   for (int j = 0; j < ny+1; j++)
+   {
+      cy = ((double) j / ny) * sy;
+      for (int i = 0; i < nx+1; i++)
+      {
+         cx = ((double) i / nx) * sx;
```

Comment:
`i` and `j` can be `double` in the loop definition, and then you won't need the cast. Additionally `cx` and `cy` can be defined in place:
```
   for (double j = 0; j < ny+1; j++)
   {
      double cy =  (j / ny) * sy;
      for (double i = 0; i < nx+1; i++)
      {
         double cx = (i / nx) * sx;
         vertices[k](0) = cx;
         vertices[k](1) = cy;
         k++;
      }
   }
```